cmd/controller/disk.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2018 OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import apis "github.com/openebs/node-disk-manager/pkg/apis/openebs.io/v1alpha1"
+
+// DiskInfo contains details of a disk which can be converted into api.Disk
+type DiskInfo struct {
+	ProbeIdentifiers ProbeIdentifier // ProbeIdentifiers contains some keys to uniquely identify each disk by probe
+	HostName         string          // HostName is hostName in which disk is attached
+	Uuid             string          // Uuid is the unique id given by ndm
+	Capacity         uint64          // Capacity is capacity of a disk
+	Model            string          // Model is model no of a disk
+	Serial           string          // Serial is serial no of a disk
+	Vendor           string          // Vendor is vendor of a disk
+	Path             string          // Path is dev path of a disk like /dev/sda
+}
+
+// ProbeIdentifiers contains some keys to uniquely identify each disk by probe
+type ProbeIdentifier struct {
+	Uuid           string // Uuid is uid of disk
```

Comment:
How is this different from the Uuid in the above DiskInfo structure?

Reply:
If we want to change disk uid generation functionality then we need this. Earlier @pawanpraka1  told we may revisit this uid generation function. As main disk discover and attach/detach event is generated by udev probe only at the time of discovering or event it generate one uid and that uid is same as disk uid that's why both are same.